### PR TITLE
[4/4] wire up transcripts to prompt mutation

### PR DESCRIPTION
**Pull request by justicart**
[4/4] wire up transcripts to prompt mutation

Summary: added Search graphql mutation, will use this to create saved searches later.
Wired up langchainAPI
Display results in Clip Search page
Added meta text under search

Test Plan:

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/bolt-foundry/bolt-foundry/pull/329).
* #343
* #342
* #341
* __->__ #329
* #328
* #324
* #322

### CODE REVIEW

File: infra/aiPlayground/langchainAPI.ts [hunk 1]
```diff
@@ -96,32 +96,32 @@ export const callAPI = async (
...
   systemMessage?: string,
   suggestedModel?: string | null | undefined,
+  documents?: string | null | undefined,
 ) => {
   let llmInterface;
   switch (suggestedModel) {
-    case "gpt-3.5-turbo":
-    case "gpt-4o-mini": {
-      llmInterface = new ChatOpenAI({
+    case "claude-3-opus-20240229":
```
Comment (randallb) [inline infra/aiPlayground/langchainAPI.ts:103]: not a fan of these magic strings. Maybe turn them into a real TS enum?

File: infra/aiPlayground/langchainAPI.ts [hunk 2]
```diff
@@ -150,13 +150,23 @@ export const callAPI = async (
...
+      return `Filename: ${sample.name}\nContent: ${content}`;
+    }).join("\n\n");
+  } else {
+    const parsedDocuments = JSON.parse(documents);
+    formattedData = parsedDocuments.map((document) => {
+      const transcript = JSON.parse(document.transcript);
+      const content = transcript.map((word) => word.word).join(" ");
+      return `Filename: ${document.filename}\nContent: ${content}`;
+    }).join("\n\n");
+  }
```
Comment (randallb) [inline infra/aiPlayground/langchainAPI.ts:169]: Would be nice not necessary to reference before / after examples in the summary or test plan maybe? I'm not clear on what was added, but it seems important?

File: packages/client/components/clipsearch/Search.tsx [hunk 3]
```diff
@@ -2,8 +2,26 @@ import { React, ReactRelay } from "deps.ts";
 import { graphql } from "packages/client/deps.ts";
 import { Input } from "packages/bfDs/Input.tsx";
 import { SearchQuery } from "packages/__generated__/SearchQuery.graphql.ts";
+import { DropdownSelector } from "packages/bfDs/DropdownSelector.tsx";
 const { useState } = React;
-const { useLazyLoadQuery } = ReactRelay;
+const { useLazyLoadQuery, useMutation } = ReactRelay;
```
Comment (randallb) [inline packages/client/components/clipsearch/Search.tsx:7]: It's a good idea to switch this to the other style import whenever you're using it.

File: packages/client/components/clipsearch/Search.tsx [hunk 4]
```diff
@@ -15,19 +33,47 @@ const query = await graphql`
...
+  setClips: (clips: string) => void;
+};
+
+export function Search({ setClips }: Props) {
+  const [commit, isInFlight] = useMutation(mutation);
   const data = useLazyLoadQuery<SearchQuery>(query, {});
   const [prompt, setPrompt] = useState("");
-  const [isInFlight, setIsInFlight] = useState(false);
+  const [clipsFound, setClipsFound] = useState<number | null>(null);
+  const [aiModel, setAiModel] = useState("gpt-4o-mini");
```
Comment (randallb) [inline packages/client/components/clipsearch/Search.tsx:45]: don't love the magic string.

File: packages/client/components/clipsearch/Search.tsx [hunk 5]
```diff
@@ -15,19 +33,47 @@ const query = await graphql`
...
+    setClipsFound(null);
     console.log("Searching for", prompt);
     console.log("Transcripts: ", data.transcripts.length);
-    setIsInFlight(true);
-    setTimeout(() => setIsInFlight(false), 3000);
+    commit({
+      variables: {
+        input: prompt,
+        // suggestedModel: "claude-3-opus-20240229",
+        // suggestedModel: "gpt-4o-mini",
```
Comment (randallb) [inline packages/client/components/clipsearch/Search.tsx:56]: what's this?

File: packages/client/components/clipsearch/Search.tsx [hunk 6]
```diff
@@ -15,19 +33,47 @@ const query = await graphql`
...
+        suggestedModel: aiModel,
+        documents: JSON.stringify(data.transcripts),
+      },
+      onCompleted: (response) => {
+        // console.log(response.searchMutation.message);
```
Comment (randallb) [inline packages/client/components/clipsearch/Search.tsx:61]: rm? or make a real logger.debug?

File: packages/client/components/clipsearch/Search.tsx [hunk 7]
```diff
@@ -40,6 +86,21 @@ export function Search() {
...
+        <DropdownSelector
+          placeholder="Select AI Model"
+          value={aiModel}
+          onChange={(value) => setAiModel(value)}
+          options={{
+            "GPT 3.5 turbo": "gpt-3.5-turbo",
+            "GPT 4o mini": "gpt-4o-mini",
+            "Claude 3 opus": "claude-3-opus-20240229",
+            "Claude 3.5 sonnet": "claude-3-5-sonnet-20240620",
+          }}
```
Comment (randallb) [inline packages/client/components/clipsearch/Search.tsx:100]: Please don't use magic objects / strings... just put this at the top somewhere as a const or something.

File: packages/client/pages/ClipSearchPage.tsx [hunk 8]
```diff
@@ -6,6 +6,7 @@ import { List } from "packages/bfDs/List.tsx";
 import { ListItem } from "packages/bfDs/ListItem.tsx";
 
 export function ClipSearchPage() {
+  const [clips, setClips] = React.useState<string>();
```
Comment (randallb) [inline packages/client/pages/ClipSearchPage.tsx:9]: We can now do { useState } from "react" etc.

File: packages/graphql/types/BfGraphQLSearch.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
+import {
+  mutationField,
+  nonNull,
+  objectType,
+  stringArg,
+} from "packages/graphql/deps.ts";
+import { GraphQLContext } from "packages/graphql/graphql.ts";
+import { callAPI } from "infra/aiPlayground/langchainAPI.ts";
+
+// Define the output type for the mutation response
```
Comment (randallb) [inline packages/graphql/types/BfGraphQLSearch.ts:10]: This seems like an unhelpful comment.

File: packages/graphql/types/BfGraphQLSearch.ts
Comment (randallb) [inline packages/graphql/types/BfGraphQLSearch.ts:1]: Search is a pretty generic type name... maybe "clip search" or "library search"?

File: packages/graphql/types/BfGraphQLSearch.ts [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+        undefined,
+        suggestedModel,
+        documents,
+      );
+      return {
+        success: true,
+        message,
+      };
+    } catch (error) {
+      console.error("Form submission error:", error);
```
Comment (randallb) [inline packages/graphql/types/BfGraphQLSearch.ts:43]: use a real logger please

File: packages/graphql/types/BfGraphQLSearch.ts [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+      return {
+        success: false,
+        message: "Form submission failed.",
```
Comment (randallb) [inline packages/graphql/types/BfGraphQLSearch.ts:46]: this is a meh content design for the message. Maybe we should start using fbt and get Rich to do content design? lols

**Review by randallb — APPROVED** [review]
A lot of stuff that could be improved but nothing absolutely wrong with any of it.

File: infra/aiPlayground/langchainAPI.ts (hunk 2 above)
Comment (justicart) [reply infra/aiPlayground/langchainAPI.ts:169]: "samples" is the test data... I left it so it runs colby's playground... however I could just put the samples in colby's playground and pass them as documents

File: packages/client/components/clipsearch/Search.tsx (hunk 5 above)
Comment (justicart) [reply packages/client/components/clipsearch/Search.tsx:56]: testing different ones before we added the dropdown selector 🤦

File: packages/graphql/types/BfGraphQLTranscript.ts [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+  description: "A transcript of a clip.",
+  definition: (t) => {
+    t.implements(BfNodeGraphQLType);
+    t.string("transcript");
+    t.string("filename");
+  },
+});
+
+export const BfGraphQLTranscriptsQuery = extendType({
+  type: 'Query',
```
Comment (justicart) [inline packages/graphql/types/BfGraphQLTranscript.ts:26]: extendType: "BfOrganization",

File: packages/graphql/types/BfGraphQLTranscript.ts [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+  definition(t) {
+    t.nonNull.list.field('transcripts', {
```
Comment (justicart) [inline packages/graphql/types/BfGraphQLTranscript.ts:28]: t.connectionField('transcripts', {

File: packages/graphql/types/BfGraphQLTranscript.ts [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+      type: BfGraphQLTranscriptType,
+      resolve: async (_, __, { bfCurrentViewer }) => {
+        logger.debug('Fetching all transcripts');
+        const transcripts = await BfTranscript.findTranscriptsByViewer(bfCurrentViewer);
```
Comment (justicart) [inline packages/graphql/types/BfGraphQLTranscript.ts:32]: const transcript = await BfTranscript.queryConnectionsForGraphQL(bfCurrentViewer, args);

File: packages/graphql/types/BfGraphQLTranscript.ts [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+        logger.debug('Fetched all transcripts successfully', transcripts);
+        return transcripts.map(transcript => transcript.toGraphql());
```
Comment (justicart) [inline packages/graphql/types/BfGraphQLTranscript.ts:34]: delete this because query connections does this

Comment (justicart) [reply packages/graphql/types/BfGraphQLTranscript.ts:34]: return transcripts;

**Pull request closed without merging**